documentation/swagger-checklist.md

```diff
@@ -43,82 +45,236 @@ This document defines a Swagger checklist for Microsoft Azure, along with clarif
 
-&#x25a2; *M3001*: Each operation described in a Swagger specification MUST be tested prior to opening a pull request against the preview branch of the [Azure REST API Specs](https://github.com/azure/azure-rest-api-specs/) GitHub repo.
+- [ ] **M2043**: A swagger specification *MUST NOT* have any operation with an HTTP verb outside this (case-insensitive) list ["DELETE", "GET", "PUT", "PATCH", "HEAD", "OPTIONS", "POST"].
+
+- [ ] **M2044**: An example specified in the spec as per the swagger specification for a response *MUST* be in the response object keyed on the "MIME-TYPE". More details can be found [here](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#exampleObject).
+
+- [ ] **M2045**: An example specified in the spec as per the swagger specification for a model in the request body *MUST* be specified in the free-form example property of that model definition. More details can be found [here](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#fixed-fields-13).
+
+- [ ] **M2046**: A parameter object *MUST* have a `"name"` property defined with a non-empty string as its value.
+
+- [ ] **M2047**: A parameter object *MUST* have an `"in"` property defined with a non-empty string. The value can be one of "query", "header", "path", "body", "formData".
+
+- [ ] **M2048**: Every operation *MUST* have a "default" status code in the "responses" section. The "default" corresponds to the error returned by the operation.
+
+- [ ] **M2049**: "externalDocs" property *MUST* have both "url" and "description" accurately provided. More details can be found [here](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#external-documentation-object).
+
+- [ ] **M2050**: Odata parameters like "$filter" *MUST* document the properties on which the object can be filtered on  and the supported operators in the description.
+
+- [ ] **M2050**: Odata parameters like "$filter" *MUST* have an externalDocs that provides good examples of how the "$filter" parameter can be used.
+
+- [ ] **M2051**: Parameters or properties that can only accept one value *MUST* be marked as "required" and they *MUST* have an "enum" array constraint with one valid value in it.
+
+- [ ] **M2052**: Every swagger spec *MUST* have a "securityDefinitions" object accurately defined. An example can be found over [here](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-analysisservices/2016-05-16/swagger/analysisservices.json#L26).
+
+- [ ] **M2053**: An operationId in the swagger spec *MUST NOT* be split by more than one underscore. Please use the "tags" array for adding extra names.
+
+- [ ] **M2054**: If the service (in Azure) only supports PATCHING tags on a Resource in the PATCH operation and has a PUT on the same resource which is named as "CreateOrUpdate", then it *MUST NOT* expose the "PATCH" operation in the swagger spec. 
```

Comment:
This recommendation is in conflict with another MUST (M1006) which states that a PUT MUST be named Create. Which means that this could never happen. If we follow all the MUSTs, that is :)